### DISCUSSION TITLE
Improve inline variable

### DISCUSSION THREAD

**Discussion by ahdung** [dotnet/csharplang#4462] [Language Ideas]
I hope this code work:
```c#
if (_companies?.TryGetValue(CompanyName, out var code) is true)
{
    CompanyCode = code; // 'code' not be initialized
}
```

instead of:
```c#
string code = default;
if (_companies?.TryGetValue(CompanyName, out code) is true)
{
    CompanyCode = code;
}
```
or
```c#
if (_companies != null && _companies.TryGetValue(CompanyName, out var code))
{
    CompanyCode = code;
}
```

**Comment by jnm2** [comment]
I do, too. This is tracked at https://github.com/dotnet/csharplang/discussions/916 and @333fred has expressed interest in championing it: https://github.com/dotnet/csharplang/issues/3365#issuecomment-615383742

Oh, more recently, a spec change has been proposed by @RikkiGibson at https://github.com/dotnet/csharplang/discussions/4240.

**Reply by RikkiGibson** [reply]
The goal is to make this happen in C# 10.
